src/overlays/actors/ovl_En_Syateki_Man/z_en_syateki_man.c

```diff
@@ -41,73 +56,1252 @@ const ActorInit En_Syateki_Man_InitVars = {
+                    this->unk_284 = 0xA3A;
+                    break;
+            }
+        }
+        this->actionFunc = func_809C6E30;
+    } else {
+        func_800B8614(&this->actor, globalCtx, 120.0f);
+    }
+
+    if (player->actor.world.pos.z < 135.0f) {
+        player->actor.world.pos.z = 135.0f;
+    }
+}
+
+void func_809C6A04(EnSyatekiMan* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (func_80147624(globalCtx)) {
+        if (globalCtx->msgCtx.choiceIndex == 0) {
+            if (!CUR_UPG_VALUE(UPG_QUIVER)) {
+                play_sound(NA_SE_SY_ERROR);
+                func_801518B0(globalCtx, 0xA30, &this->actor);
+                this->unk_284 = 0xA30;
+            } else if (gSaveContext.rupees < 20) {
+                globalCtx = globalCtx;
+                play_sound(NA_SE_SY_ERROR);
+                func_801518B0(globalCtx, 0xA31, &this->actor);
+                this->unk_284 = 0xA31;
+                if (this->unk_26A == 4) {
+                    gSaveContext.minigameState = 3;
+                }
+                this->unk_26A = 3;
+            } else {
+                func_8019F208();
+                func_801159EC(-20);
+                gSaveContext.weekEventReg[63] |= 1;
+                gSaveContext.weekEventReg[63] &= (u8)~2;
+                globalCtx->msgCtx.unk11F22 = 0x43;
+                globalCtx->msgCtx.unk12023 = 4;
+                this->unk_26A = 7;
+                player->stateFlags1 |= 0x20;
+                this->actionFunc = func_809C7FFC;
+            }
+        } else {
+            func_8019F230();
+
+            switch (CURRENT_DAY) {
+                case 1:
+                    func_801518B0(globalCtx, 0xA2D, &this->actor);
+                    this->unk_284 = 0xA2D;
+                    break;
+
+                case 2:
+                    func_801518B0(globalCtx, 0xA2E, &this->actor);
+                    this->unk_284 = 0xA2E;
+                    break;
+
+                case 3:
+                    func_801518B0(globalCtx, 0xA2F, &this->actor);
+                    this->unk_284 = 0xA2F;
+                    break;
+            }
+
+            if (this->unk_26A == 4) {
+                gSaveContext.minigameState = 3;
+            }
+
+            this->unk_26A = 3;
+        }
+    }
+}
+
+void func_809C6C2C(EnSyatekiMan* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (func_80147624(globalCtx)) {
+        switch (this->unk_284) {
+            case 2600:
+            case 2601:
+                func_801518B0(globalCtx, 0xA2A, &this->actor);
+                this->unk_284 = 0xA2A;
+                break;
+
+            case 2603:
+            case 2604:
+            case 2613:
+                globalCtx->msgCtx.unk11F22 = 0x43;
+                globalCtx->msgCtx.unk12023 = 4;
+                player->actor.freezeTimer = 0;
+                func_80112AFC(globalCtx);
+                globalCtx->interfaceCtx.hbaAmmo = 0x50;
```

Comment:
```suggestion
                globalCtx->interfaceCtx.hbaAmmo = 80;
```
Ammo seems like it should be in dec